source/parsed_grid_generator.cc

```diff
@@ -232,62 +232,146 @@ ParsedGridGenerator<dim, spacedim>::serial()
+      {
+
+        GridGenerator::hyper_ball( tria,
+                                   point_option_one,
+                                   double_option_one);
+        return true;
+      }
+    else if (grid_name == "subhyperrectangle")
+      {
+
+        std::vector<unsigned int> refinement(dim);
+        for (unsigned int i=0; i<dim; ++i)
+          refinement[i] = un_int_option_one;
+        GridGenerator::subdivided_hyper_rectangle ( tria,
+                                                    refinement,
+                                                    point_option_two,
+                                                    point_option_one);
+        return true;
+      }
+    else if (grid_name == "hyper_shell")
+      {
+        GridGenerator::hyper_shell( tria,
+                                    point_option_one,
+                                    double_option_two,
+                                    double_option_one
+                                    /* const unsigned int   n_cells = 0,
+                                    bool  colorize = false */
+                                  );
+        return true;
+      }
+    return false;
+  }
+}
+
 template <int dim, int spacedim>
 void ParsedGridGenerator<dim, spacedim>::create(Triangulation<dim,spacedim> &tria)
 {
   Assert(grid_name != "", ExcNotInitialized());
-  if (grid_name == "rectangle")
-    {
-      GridGenerator::subdivided_hyper_rectangle (tria, un_int_vec_option_one,
-                                                 point_option_two, point_option_one,
-                                                 bool_option_one);
-    }
-  // TO BE DONE WHEN POSSIBLE
-  // else if(grid_name == "unit_hyperball")
-  // {
-  //   Point<spacedim> center;
-  //   double radius=1.;
-  //
-  //   GridGenerator::hyper_ball (tria,
-  //                               center, radius);
-  //   }
-  // else if(grid_name == "subhyperrectangle"){
-  //   if(dim == spacedim)
-  //   {
-  //
-  //   std::vector<unsigned int> refinement(spacedim);
-  //   for(unsigned int i=0; i<spacedim;++i)
-  //     refinement[i] = un_int_option_one;
-  //   GridGenerator::subdivided_hyper_rectangle (tria, refinement,
-  //                                    point_option_two, point_option_one);
-  //   }
-  //   else
-  //     Assert(true, ExcInternalError("dim != spacedim not supported"))
-  //
-  // }
-  else if (grid_name == "file")
-    {
-      GridIn<dim, spacedim> gi;
-      gi.attach_triangulation(tria);
-
-      std::ifstream in(input_grid_file_name.c_str());
-      AssertThrow(in, ExcIO());
-
-      std::string ext = extension(input_grid_file_name);
-      if (ext == "vtk")
-        gi.read_vtk(in);
-      else if (ext == "msh")
-        gi.read_msh(in);
-      else if (ext == "ucd" || ext == "inp")
-        gi.read_ucd(in);
-      else if (ext == "unv")
-        gi.read_unv(in);
-      else
-        Assert(false, ExcNotImplemented());
-    }
-  else
+  bool status = false;
+
```

Comment:
From a software design point of view, this implementation is bogus...

Call `create_grid`, and `create_grid` internally calls `default_create_grid`. No boolean returned. `default_create_grid` should fail if anything goes wrong.